### PR TITLE
Add greetings as JSON to be printed in index.html after button press

### PR DESCRIPTION
**Pull request by margaretchan**
Primary Reviewer: Sam
Secondary Reviewer: Derrill

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -24,9 +27,21 @@
...
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-  }
+    private List<String> greetings;
+
+    @Override
+    public void init(){
+        greetings = new ArrayList<>();
```
Comment (LiRq95) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: nit: Can instantiate in 1 line with = Arrays.asList("str1", "str2", ...);
Also, if this list will never change, consider using an ImmutableList instead. You can reassign it but it ensures that there can be no further mutations and is considered good practice. You could then probably make it a static final if you really decide to never change or reassign it.

Comment (margaretchan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: I'm planning on using this arraylist to store user comments in the future (in the next pull request) which needs mutability. I agree that for this use case, an immutable list would be more appropriate. I'll change the instantiation to 1 line but should I also change the data type?

Comment (LiRq95) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: Just making the instantiation 1 line is good for now. If you're going to store comments later, remember Derrill's advice on not keeping state on the class due to thread conflicts, so you'll want to get rid of the class variable at that point and probably move it into a method.

Comment (margaretchan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Fixed. By move into a method do you mean making getter and setter methods for mutating the ArrayList or moving the whole servlet class into a method?

Comment (LiRq95) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: So as mentioned, you won't want the Arraylist to be a variable in the class itself anymore as that would introduce state to the class. Instead you'll be putting a List copy in something like your postComment method where there won't be thread conflicts.

Dante and Josef are currently in the middle of implementing using the CloudStorage to store comments. I'm not sure if you have access to their repositories, but taking a peek at their code should shed some light as to what I mean for the next steps :)

**Pull request merged**